examples/NetCore2/ConsoleExample/Program.cs

```diff
@@ -44,21 +44,20 @@ private static IServiceProvider BuildDi()
             // Runner is the custom class
             services.AddTransient<Runner>();
 
-            services.AddSingleton<ILoggerFactory, LoggerFactory>();
-            services.AddSingleton(typeof(ILogger<>), typeof(Logger<>));
-            services.AddLogging(builder => builder.SetMinimumLevel(LogLevel.Trace));
+            var config = new ConfigurationBuilder()
+                .SetBasePath(System.IO.Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .Build();
 
-            var serviceProvider = services.BuildServiceProvider();
-
-            var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
-
-            // configure NLog
-            loggerFactory.AddNLog(new NLogProviderOptions
+            // configure Logging with NLog
+            services.AddLogging(loggingBuilder =>
             {
-                CaptureMessageTemplates = true,
-                CaptureMessageProperties = true
+                loggingBuilder.ClearProviders();
+                loggingBuilder.SetMinimumLevel(Microsoft.Extensions.Logging.LogLevel.Trace);
+                loggingBuilder.AddNLog(config);
```

Comment:
I'm not sure if this is DI-proof. 

I think it's a good change to move to the Factory instead of Provider, that's the one we could also bootstrap from DI. See https://github.com/NLog/NLog.Extensions.Logging/pull/253

Reply:
Please explain DI-proof?

Reply:
I don't like to change the call, if we need a new dependency. But is a bit debatable if the config is here an dependency or not.

Reply:
I was planning that NLogLoggingProvider.Options and NLog-LoggingConfiguration was loaded from IConfiguration. So it would be natural to receive the config when doing AddNLog.

But not that skilled with DI and fluent-interface, so please help with creating a great interface.